akka-docs/src/main/paradox/stream/operators/Source/unfoldResource.md

```diff
@@ -14,14 +14,52 @@ Wrap any resource that can be opened, queried for next element (in a blocking wa
 
 ## Description
 
-Wrap any resource that can be opened, queried for next element (in a blocking way) and closed using three distinct functions into a source.
+`Source.unfoldResource` allows us to safely extract stream elements from blocking resources by providing it with three functions: 
+
+1. `create`: Open or create the resource
+1. `read`: Fetch the next element or signal that we reached the end of the stream by returning a @java[`Optional.empty`]@scala[`None`]
+1. `close`: Close the resource, invoked on end of stream or if the stream fails
+
+The functions are by default called on Akka's dispatcher for blocking IO to avoid interfering with other stream operations. 
+See @ref:[Blocking Needs Careful Management](/typed/dispatchers.md#blocking-needs-careful-management) for an explanation on why this is important.
```

Comment:
Apparently paradox doesn't support absolute links like this. Even if it did, it's questionable whether they should be interpreted relative to the paradox root or to the project repo root: for snippets absolute links are relative to the project repo root, not only because that way you can include snippets from files outside the documentation tree, but also because that way the links can also be followed when browsing the markdown files on github.

Anyway at least for now:

```suggestion
See @ref:[Blocking Needs Careful Management](../../../typed/dispatchers.md#blocking-needs-careful-management) for an explanation on why this is important.
```

Reply:
I believe I misguided @johanandren here...